Composer/packages/extensions/visual-designer/src/components/menus/EdgeMenu.tsx

```diff
@@ -26,12 +31,44 @@ interface EdgeMenuProps {
   addCoachMarkRef?: (ref: { [key: string]: HTMLDivElement }) => void;
 }
 
+const buildCustomActionsSubmenu = (
+  label: string,
+  customizedActionGroups: any[][],
+  handleType,
+  factory
+): IContextualMenuItem => {
+  const itemGroups: IContextualMenuItem[][] = customizedActionGroups
+    .filter(actionGroup => Array.isArray(actionGroup) && actionGroup.length)
+    .map(actionGroup => {
+      return actionGroup.map(({ title: $kind }) => ({
+        key: $kind,
+        name: $kind,
+        $kind,
```

Comment:
This should be wrapped in a `data` object to conform to the menu item type.

Reply:
After the refactoring work, the EdgeMenu only emits the key of the item.
Usually, the emitted key is set to the action `$kind`.

The motivation of doing so is to avoid coupling `DialogFactory` in menu components